### DISCUSSION TITLE
Application start

### DISCUSSION THREAD

**Discussion by bpecsek** [rabbibotton/clog#22] [General]
Hi David!

First of all I love Clog.
Thanks for the opportunity to try it in this early stage of development.

I am playing with the tutorials and demos and realized that a new app always opens in a new browser tab.

Is there a way to open an app in a new window, hiding that it is a browser at all instead of in a browser tab?

Like demo 3 starts in a browser tab. Is there a way to run it to look like a desktop application?

Regards,
Bela

**Comment by rabbibotton** [comment]
Once upon a time you could do a pop-up but pop-up is a dirty work now :)
The easiest way to what you want is use Ceramic, when I can will do a tutorial for it. I made the snake demo into a native Mac app in 10min and a few lines of code :)

Important note the demos and tutorials purposely ask the os to open a browser - That is not an automatic behavior at all, the normal way is to browse to your clog server - http://127.0.0.1:8080 (or whatever you set. I test the iPhone and other devices all the time by opening to the ip for my laptop instead.

What this means is any "browser" or "browser control" you design can be the frontend local or remote.

**Reply by bpecsek** [reply]
I fully understand your reasoning and agree with you generally.
For most programs, this is the best approach indeed to run the software inside the browser as a web application and run it locally or remotely.
However, sometimes a desktop one is the best way to go.
As you indicated it can be done using Ceramic, however, to give me some speed (I am still figuring out clog) I would appreciate it if you could share the code, when your time allows, on how you made the snake demo into a Mac app.

**Comment by rabbibotton** [comment]
Maybe you can write the tutorial, here is link to  http://ceramic.github.io/  it is in quickslip and works well with CLOG.
I can also help if you want to show how to write bindings to its javascript APIs like for menus etc. if want to help things grow faster :)

**Reply by bpecsek** [reply]
I am still learning (relearning) CL and Clog and Ceramic so I am not the best person yet to write the tutorial.
Maybe later when I'll have gained a solid grasp about them. I would surely do it.

**Comment by rabbibotton** [comment]
"early stage of development" - I wouldn't per se call early as the tech in my Ada bindings running production many years, in fact I would say that wile system isn't load tested yet for large websites, I would be confident to write and ship an app in it or small website :) (the web tech is also very stable in the CL components so probably being too cautious.)

**Reply by bpecsek** [reply]
I am sorry David for my incorrect wording. I can see that your work behind this development is immense.
I referred to Clog itself as "early stage" not the hard work behind it.

**Reply by rabbibotton** [reply]
No issues at all :) I will get to those tutorials and feature sets in time :)
